src/kakarot/library.cairo

```diff
@@ -170,33 +168,35 @@ namespace Kakarot {
     }
 
     // @notice deploy contract account
-    // @dev Deploys a new starknet contract which functions as a new contract account and
-    //      will be mapped to an evm address
-    // @param bytes_len: the contract bytecode lenght
-    // @param bytes: the contract bytecode
+    // @dev First deploy a contract_account with no bytecode, then run the calldata as bytecode with the new address,
+    //      then set the bytecode with the result of the initial run
+    // @param bytecode_len: the contract bytecode lenght
+    // @param bytecode: the contract bytecode
     // @return evm_contract_address The evm address that is mapped to the newly deployed starknet contract address
     // @return starknet_contract_address The newly deployed starknet contract address
-    @external
-    func deploy_contract{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        bytes_len: felt, bytes: felt*
-    ) -> (evm_contract_address: felt, starknet_contract_address: felt) {
+    func deploy{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(bytecode_len: felt, bytecode: felt*) -> (
+        evm_contract_address: felt, starknet_contract_address: felt
+    ) {
         alloc_locals;
         let (current_salt) = salt.read();
         let (class_hash) = evm_contract_class_hash.read();
 
+        // Prepare constructor data
         let (local calldata: felt*) = alloc();
         let (kakarot_address) = get_contract_address();
-
-        // Prepare constructor data
         assert [calldata] = kakarot_address;
-        assert [calldata + 1] = bytes_len;
-        memcpy(dst=calldata + 2, src=bytes, len=bytes_len);
+        assert [calldata + 1] = 0;
 
-        // Deploy contract account
-        let (contract_address) = deploy(
+        // Deploy contract account with no bytecode
+        let (contract_address) = deploy_syscall(
```

Comment:
do we no longer emit an event with the contract_address (starknet) and evm_contract_address

Reply:
just forgot to emit, good catch